### DISCUSSION TITLE
How to dynamic append into the []excelize.RichTextRun

### DISCUSSION THREAD

**Discussion by joojoooe** [qax-os/excelize#1754] [Q&A]
I want to use SetCellRichText, but I don't know how to dynamic append the json into the []excelize.RichTextRun. Thank you

**Comment by xuri** [comment]
You need to transform from JSON to Go data structure and create each excelize.RichTextRun in your way. What does your JSON string look like? What are the transform rules? That's not clear about this in your description.

**Reply by joojoooe** [reply]
type RichTextRun struct {
    Text string
}

tt := []excelize.RichTextRun{}
tt = append(tt, RichTextRun{...})

I will got the error message
cannot use RichTextRun{} (value of type RichTextRun) as excelize.RichTextRun value in argument to append

how to append?

**Reply by xuri** [reply]
Using

```go
tt = append(tt, excelize.RichTextRun{...})
```

instead of

```go
tt = append(tt, RichTextRun{...})
```

**Reply by joojoooe** [reply]
It's work. Thank you
